# HG changeset patch
# User Ludovic Chabant <ludovic@chabant.com>
# Date 1583871438 0
#      Tue Mar 10 20:17:18 2020 +0000
# Node ID 70878db5d0ede9c39128328cfa8cb6ee758d5374
# Parent  a08bbdf839ae08ff69c6f99a289ee85b3a012f03
archive: fix crash when archiving to gzip file with Python 3.8.2+

Differential Revision: https://phab.mercurial-scm.org/D8272

diff --git a/mercurial/archival.py b/mercurial/archival.py
--- a/mercurial/archival.py
+++ b/mercurial/archival.py
@@ -146,7 +146,7 @@
                 self.timestamp = timestamp
             gzip.GzipFile.__init__(self, *args, **kw)
 
-        def _write_gzip_header(self):
+        def _write_gzip_header(self, *args, **kwargs):
             self.fileobj.write(b'\037\213')  # magic header
             self.fileobj.write(b'\010')  # compression method
             fname = self.name
